src/Type/Symfony/ExtensionGetConfigurationReturnTypeExtension.php

```diff
@@ -0,0 +1,72 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type\Symfony;
+
+use PhpParser\Node\Expr\MethodCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\MethodReflection;
+use PHPStan\Reflection\ReflectionProvider;
+use PHPStan\Type\DynamicMethodReturnTypeExtension;
+use PHPStan\Type\NullType;
+use PHPStan\Type\ObjectType;
+use PHPStan\Type\Type;
+use function count;
+use function str_contains;
+use function strrpos;
+use function substr_replace;
+
+class ExtensionGetConfigurationReturnTypeExtension implements DynamicMethodReturnTypeExtension
+{
+
+	/** @var ReflectionProvider */
+	private $reflectionProvider;
+
+	public function __construct(ReflectionProvider $reflectionProvider)
+	{
+		$this->reflectionProvider = $reflectionProvider;
+	}
+
+	public function getClass(): string
+	{
+		return 'Symfony\Component\DependencyInjection\Extension\Extension';
+	}
+
+	public function isMethodSupported(MethodReflection $methodReflection): bool
+	{
+		return $methodReflection->getName() === 'getConfiguration'
+			&& $methodReflection->getDeclaringClass()->getName() === 'Symfony\Component\DependencyInjection\Extension\Extension';
+	}
+
+	public function getTypeFromMethodCall(
+		MethodReflection $methodReflection,
+		MethodCall $methodCall,
+		Scope $scope
+	): ?Type
+	{
+		$extensionType = $scope->getType($methodCall->var);
+		$classes = $extensionType->getObjectClassNames();
+		if (count($classes) !== 1) {
+			return null;
+		}
+
+		$extensionName = $classes[0];
+		if (str_contains($extensionName, "\0")) {
+			return new NullType();
+		}
+
+		$lastBackslash = strrpos($extensionName, '\\');
+		if ($lastBackslash === false) {
+			return new NullType();
+		}
+
+		$configurationName = substr_replace($extensionName, '\Configuration', $lastBackslash);
+		if (!$this->reflectionProvider->hasClass($configurationName)) {
+			return new NullType();
+		}
+
+		// TODO: return NullType if configuration class has constructor with required arguments
```

Comment:
Please solve this todo, or remove it.

Reply:
If you can provide some pointers on how to determine this I will try to implement it :) I haven't been able to find some example on how to determine / inspect such information.

Reply:
You can get ClassReflection from ReflectionProvider and ask about constructor from there.